### PR TITLE
Trello 73 - Comment Section

### PR DESCRIPTION
**Pull request by Dragonite**
AJAX for basic commenting.

### CODE REVIEW

File: core/static/js/viewCase.js [hunk 1]
```diff
@@ -75,4 +77,47 @@ $("#submit_response").click(function () {
...
+      url: '/cases/ajax/submit_comment/' + id,
+      dataType: 'json',
+      data: {
+        'comment_body': comment_body,
+        'comment_is_anon': comment_is_anon
+      },
+      success: function (data) {
+        let date = moment(data.comment.date).format("MMM D YYYY, hh:mm a.");
+        let name = data.comment.is_anon ? data.user.name + " (Anonymous)" : data.user.name;
+        htmlstring =
```
Comment (tim-ings) [inline core/static/js/viewCase.js]: Clear the comment input on success

File: core/static/js/viewCase.js [hunk 2]
```diff
@@ -75,4 +77,47 @@ $("#submit_response").click(function () {
...
+
+$('#submit_comment').click(function () {
+  let comment_body = document.getElementById('comment-box').value;
+  let comment_is_anon = document.getElementById('is_anonymous_checkbox').checked;
+  let id = document.getElementById('case_id').innerText;
+  if (comment_body) {
+    $.ajax({
+      url: '/cases/ajax/submit_comment/' + id,
+      dataType: 'json',
+      data: {
```
Comment (tim-ings) [inline core/static/js/viewCase.js]: This should have method POST

File: core/static/js/viewCase.js [hunk 3]
```diff
@@ -75,4 +77,47 @@ $("#submit_response").click(function () {
...
+        'comment_body': comment_body,
+        'comment_is_anon': comment_is_anon
```
Comment (tim-ings) [inline core/static/js/viewCase.js]: Dont need these "comment_" prefixes

File: core/case_study/views.py [hunk 4]
```diff
@@ -230,6 +239,40 @@ def validate_answer(request, case_study_id):
...
+        'feedback': case.feedback,
+        'comments': comments_json
     }
     return JsonResponse(data)
+
+def submit_comment(request, case_study_id):
+    case = get_object_or_404(CaseStudy, pk=case_study_id)
+    comment_body = request.GET.get('comment_body', None)
+    comment_is_anon = request.GET.get('comment_is_anon', None).capitalize()
+    print(request.user.first_name)
```
Comment (tim-ings) [inline core/case_study/views.py]: Remove this

File: core/case_study/views.py [hunk 5]
```diff
@@ -230,6 +239,40 @@ def validate_answer(request, case_study_id):
...
+            'name': request.user.get_full_name(),
+            'is_staff': request.user.is_staff
+        }
+    }
+    return JsonResponse(data)
+
+
+def show_comments(request, case_study_id):
+    case = get_object_or_404(CaseStudy, pk=case_study_id)
+    user_attempts = len(Attempt.objects.filter(case_study=case, user=request.user))
```
Comment (tim-ings) [inline core/case_study/views.py]: Use inbuilt django count

File: core/static/js/viewCase.js [hunk 6]
```diff
@@ -75,4 +77,47 @@ $("#submit_response").click(function () {
...
+      },
+      success: function (data) {
+        let date = moment(data.comment.date).format("MMM D YYYY, hh:mm a.");
```
Comment (tim-ings) [inline core/static/js/viewCase.js]: Do we need moment for this one line

File: core/static/js/viewCase.js [hunk 7]
```diff
@@ -75,4 +77,47 @@ $("#submit_response").click(function () {
...
+
+$('#discussion').toggle();
+$('#show_discussion').click(function () {
+  $('#discussion').slideToggle();
+  $('#show_discussion').slideToggle();
+})
+
+$('#submit_comment').click(function () {
+  let comment_body = document.getElementById('comment-box').value;
+  let comment_is_anon = document.getElementById('is_anonymous_checkbox').checked;
```
Comment (tim-ings) [inline core/static/js/viewCase.js]: The is_anon checkbox doesnt do anything

File: core/static/js/viewCase.js [hunk 8]
```diff
@@ -75,4 +77,47 @@ $("#submit_response").click(function () {
...
+                      <div class="d-flex w-100 justify-content-between">\
+                        <small class="text-muted">\
+                          ${name}\
+                        </small>\
+                        <small class="text-muted">${date}</small>\
+                      </div>\
+                      <p class="mb-1">${data.comment.body}</p>\
+                    </div>\
+                  </div>\
+                </div>`;
```
Comment (tim-ings) [inline core/static/js/viewCase.js]: Username and comment date should be split onto seperate lines when the screen is narrow. Currently they are very cramped on mobile.

File: core/case_study/urls.py [hunk 9]
```diff
@@ -8,5 +8,7 @@
     path("start-new-case/", views.start_new_case, name="start-new-case"),
     path("create-new-case/<int:case_study_id>", views.create_new_case, name="create-new-case"),
     path("view/<int:case_study_id>/", views.view_case, name="view-case"),
-    path("ajax/validate_answer/<int:case_study_id>/", views.validate_answer, name="validate-answer")
+    path("ajax/validate_answer/<int:case_study_id>/", views.validate_answer, name="validate-answer"),
+    path("ajax/view_comments/<int:case_study_id>/", views.show_comments, name="show-comments"),
+    path("ajax/submit_comment/<int:case_study_id>/", views.submit_comment, name="submit-comment"),
```
Comment (tim-ings) [inline core/case_study/urls.py]: These endpoints should be 

- GET /api/v1/comments?post=12 -> returns all comments for post 12
- POST /api/v1/comments -> body is json with comment data
- etc.

**Comment by Dragonite** [comment]
All changes fixed. Moment.js used as it provides a very similar format to Django's time formatting. Minified version is very small as well. Anon does work, own messages are displayed by `Name (Anonymous)` if anonymous. Other users are displayed as `Anonymous`.

**Pull request merged**